tests/system/small/test_numpy.py

```diff
@@ -70,6 +70,13 @@ def test_df_ufuncs(scalars_dfs, opname):
     ).to_pandas()
     pd_result = getattr(np, opname)(scalars_pandas_df[["float64_col", "int64_col"]])
 
+    # In NumPy versions 2 and later, `np.floor` and `np.ceil` now produce integer
+    # outputs for the "int64_col" column.
+    if opname in ["floor", "ceil"] and isinstance(
+        pd_result["int64_col"].dtypes, pd.Int64Dtype
+    ):
```

Comment:
Seems like a big behavior change?! I thought numpy 2 was backwards compatible?

Reply:
There were behavior changes. See: https://numpy.org/doc/stable/numpy_2_0_migration_guide.html#numpy-2-migration-guide

I suspect it may relate to this change: https://numpy.org/neps/nep-0050-scalar-promotion.html#nep50

Type promotion is no longer value dependent, only dtype dependent.

Reply:
Should we match the new behavior then? dtype-dependent is something we can actually emulate perfectly, unlike value-dependency

Reply:
> Should we match the new behavior then? dtype-dependent is something we can actually emulate perfectly, unlike value-dependency

Yes. Maybe bigframes 2.0? Could you file an issue @TrevorBergeron ?